spec-tests/src/test/java/org/candlepin/spec/owners/OwnerResourcePoolFilterSpecTest.java

```diff
@@ -0,0 +1,254 @@
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.owners;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.candlepin.spec.bootstrap.assertions.StatusCodeAssertions.assertBadRequest;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.candlepin.dto.api.client.v1.ActivationKeyDTO;
+import org.candlepin.dto.api.client.v1.AttributeDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.dto.api.client.v1.UserDTO;
+import org.candlepin.invoker.client.ApiException;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.data.builder.ActivationKeys;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.ProductAttributes;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@SpecTest
+public class OwnerResourcePoolFilterSpecTest {
+
+    @Test
+    public void shouldListPoolsWithMatchesAgainstProvidedProducts() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = createOwner(adminClient);
+        ApiClient userClient = createUserClient(adminClient, owner);
+
+        ConsumerDTO consumer = userClient.consumers().createConsumer(Consumers.random(owner));
+        String providedName = "test_name";
+        Set<ProductDTO> providedProducts = new HashSet<>();
+        providedProducts.add(createProductByOwner(adminClient, owner, Products.random().name(providedName)));
+        providedProducts.add(createProductByOwner(adminClient, owner, Products.random()));
+        providedProducts.add(createProductByOwner(adminClient, owner, Products.random()));
+        ProductDTO product = createProductByOwner(
+            adminClient, owner, Products.random().providedProducts(providedProducts));
+        adminClient.owners().createPool(owner.getKey(), Pools.random(product));
+
+        List<PoolDTO> pools = adminClient.owners().listOwnerPools(owner.getKey(), consumer.getUuid(), null,
+            null, null, null, null, List.of(providedName), null,
+            null, null, null, null, null, null, null, null);
+        assertEquals(1, pools.size());
+        assertEquals(owner.getKey(), pools.get(0).getOwner().getKey());
```

Comment:
Since we're targeting stuff off the first pool of the list, we could extract it and clean up the repetition and UuT fetching a bit:

```java

assertThat(pools)
  .isNotNull()
  .hasSize(1);

PoolDTO pool = pools.get(0);
assertThat(pool)
  .isNotNull()
  .returns(product.getId(), PoolDTO::getProductId)
  .extracting(PoolDTO::getProvidedProducts)
  .isNotNull()
  .hasSize(3);

assertThat(pool.getOwner())
  .isNotNull()
  .returns(owner.getKey(), OwnerDTO::getKey);
```

Reply:
done